### PR TITLE
Parse concatenated .wasm files.

### PR DESCRIPTION
**Pull request by mbebenita**
Would be nice to parse `.wasm` files that are made up of many concatenated `.wasm` files. This PR changes the parser to check for the `.wasm` header when parsing a new section.

The large `spec.wasm` file in this PR is generated from the SpiderMonkey `spec.js` harness when executing the wasm spec tests.

### CODE REVIEW

**Comment by mbebenita** [comment]
I don't see how this would work, I still need to emit initial so I can consume the wasm header.

File: src/WasmParser.ts [hunk 1]
```diff
@@ -808,6 +821,15 @@ export class BinaryReader {
       this.state = BinaryReaderState.END_WASM;
       return true;
     }
+    if (this._pos < this._length - 4) {
```
Comment (yurydelendik) [inline src/WasmParser.ts]: more complex check is needed when _eof == false, `if (this._pos < this._length - (this._eof ? 4 : 1)) {`, then `if (magicNumber === (WASM_MAGIC_NUMBER& ((1 << (Math.min(4, this._length - this._pos) * 8)) - 1)))`

File: src/WasmDis.ts [hunk 2]
```diff
@@ -87,9 +87,16 @@ export class WasmDisassembler {
       if (!reader.read())
         return null;
       switch (reader.state) {
+        case BinaryReaderState.INITIAL:
```
Comment (yurydelendik) [inline src/WasmDis.ts]: nit: remove unreachable state

File: src/WasmParser.ts [hunk 3]
```diff
@@ -903,16 +925,28 @@ export class BinaryReader {
...
         }
         this.result = {magicNumber: magicNumber, version: version};
         this.state = BinaryReaderState.BEGIN_WASM;
         return true;
       case BinaryReaderState.END_WASM:
+        this.result = null;
+        this.state = BinaryReaderState.BEGIN_WASM;
+        if (this.hasMoreBytes()) {
+          this.state = BinaryReaderState.INITIAL;
+          return true;
```
Comment (yurydelendik) [inline src/WasmParser.ts]: return this.read();

**Pull request merged**